### PR TITLE
Add --cve Flag to Safety Scan Command for CVE Details Reporting

### PR DESCRIPTION
**Pull request by dylanpulver**
This PR introduces a new `--cve` flag to the safety scan command, enabling users to fetch and display detailed CVE (Common Vulnerabilities and Exposures) information for identified vulnerabilities in their project dependencies.
Changes Made

**1. New --cve Flag**
- Added the --cve flag to the safety scan command.
- When enabled, the command fetches detailed CVE data for vulnerabilities and displays the information in JSON format.

**2. CVE Data Extraction**
- Extracted relevant CVE details for each vulnerability, including:
   - package: The affected package name.
   - vulnerability_id: The unique safety identifier for the vulnerability.
   - CVE: CVE identifiers and associated metadata (e.g., severity scores, vector strings).
   - more_info: Safety URL linking to additional vulnerability details.
   -  advisory: A brief description of the vulnerability.
   - severity: The severity level of the vulnerability (CRITICAL, HIGH, MEDIUM, LOW, UNKNOWN).

**3. Custom Severity-Based Sorting**
-  Sorted CVE data by severity in descending order using a custom mapping:
   - CRITICAL > HIGH > MEDIUM > LOW > UNKNOWN.

**4. Enhanced Output**
- If CVE details are found:
   - Output is displayed in JSON format for easier parsing and integration.
- If no CVE details are found:
   - Displays a user-friendly message indicating no CVE information is available.

**5. Fallback Handling for Missing/Unknown Data**
- Added default handling for cases where CVE data or severity levels are missing, ensuring graceful degradation.

Example Usage
Run the following command to scan a project and include CVE details in the output:
`safety scan --cve`

## Summary by CodeRabbit

- **New Features**
	- Introduced a new command-line option `--cve` for scanning, allowing users to request CVE details during the scan.
	- Enhanced reporting to include CVE information based on the new option.

- **Documentation**
	- Updated help text for scanning commands to clarify the inclusion of CVE details and the functionality of package updates.

### CODE REVIEW

**Comment by dylanpulver** [comment]
Tested locally by running `python -m safety scan --cve`

Data output looks like the following:
```
Fetching CVE details...

CVE Details:

{
  "cve_details": [
    {
      "package": "django",
      "affected_version": "==2.2",
      "safety_vulnerability_id": "49733",
      "CVE": [
        "CVE-2022-34265",
        {
          "base_score": 7.5,
          "impact_score": 6.4,
          "vector_string": "AV:N/AC:L/Au:N/C:P/I:P/A:P"
        },
        {
          "base_score": 9.8,
          "impact_score": 5.9,
          "base_severity": "CRITICAL",
          "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H"
        }
      ],
      "more_info": "https://data.safetycli.com/v/49733/eda",
      "advisory": "Django 3.2.14 and 4.0.6 include a fix for CVE-2022-34265: Potential SQL injection via Trunc(kind) and Extract(lookup_name) arguments.\r\nhttps://www.djangoproject.com/weblog/2022/jul/04/security-releases",
      "severity": "CRITICAL"
    },
    {
      "package": "django",
      "affected_version": "==2.2",
      "safety_vulnerability_id": "37815",
      "CVE": [
        "CVE-2020-7471",
        {
          "base_score": 7.5,
          "impact_score": 6.4,
          "vector_string": "AV:N/AC:L/Au:N/C:P/I:P/A:P"
        },
        {
          "base_score": 9.8,
          "impact_score": 5.9,
          "base_severity": "CRITICAL",
          "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H"
        }
      ],
      "more_info": "https://data.safetycli.com/v/37815/eda",
      "advisory": "Django 1.11.28, 2.2.10 and 3.0.3 include a fix for CVE-2020-7471: SQL Injection if untrusted data is used as a StringAgg delimiter (e.g., in Django applications that offer downloads of data as a series of rows with a user-specified column delimiter). By passing a suitably crafted delimiter to a contrib.postgres.aggregates.StringAgg instance, it was possible to break escaping and inject malicious SQL.",
      "severity": "CRITICAL"
    },
...
```

**Review by SafetyQuincyF — COMMENTED** [review]
Tested locally it works for scan but, I didn't see it in safety scan --help.
image

It populates now, I had to restart my docker container.

File: safety/scan/constants.py [hunk 1]
```diff
@@ -83,6 +83,7 @@
     "[bold]Example: safety scan --output json[/bold]"
 SCAN_SAVE_AS_HELP = "In addition to regular output save the scan results to a json, html, text, or spdx file using: FORMAT FILE_PATH\n\n" \
     "[bold]Example: safety scan --save-as json results.json[/bold]"
+SCAN_CVE_HELP = "Include CVE details in the output."
```
Comment (SafetyQuincyF) [inline safety/scan/constants.py:86]: Just for consistency.

SCAN_CVE_HELP = "Include CVE details in the output\n\n" \
    "[bold]Example: safety scan --cve[/bold]"

**Comment by dylanpulver** [comment]
> Tested locally it works for scan but, I didn't see it in safety scan --help. image
> 
> It populates now, I had to restart my docker container.

Sounds good! I have it there:
image

**Comment by SafetyQuincyF** [comment]
for the launch.json in .vscode can you add the below please?
```   {
            "name": "Safety Scan --help",
            "type": "debugpy",
            "request": "launch",
            "module": "safety",
            "args": [
                "scan",
                "--help"
            ],
            "console": "integratedTerminal"
        },
        {
            "name": "Safety Scan --cve",
            "type": "debugpy",
            "request": "launch",
            "module": "safety",
            "args": [
                "scan",
                "--cve"
            ],
            "console": "integratedTerminal"
        },

File: safety/scan/constants.py (hunk 1 above)
Comment (dylanpulver) [reply safety/scan/constants.py]: awesome

**Pull request closed without merging**